### PR TITLE
add: segmentation guide.

### LINKED ISSUE #5181: Add a guide for semantic segmentation
Currently, we have these guides for object detection and image classification:

* https://huggingface.co/docs/datasets/object_detection
* https://huggingface.co/docs/datasets/image_classification

I am proposing adding a similar guide for semantic segmentation. 

I am happy to contribute a PR for it. 

Cc: @osanseviero @nateraw

### PR DESCRIPTION
**Pull request by sayakpaul**
Closes #5181 

I have opened a PR on Hub (https://huggingface.co/datasets/huggingface/documentation-images/discussions/5) to include the images in our central Hub repository. Once the PR is merged I will edit the image links. 

I have also prepared a [Colab Notebook](https://colab.research.google.com/drive/1BMDCfOTBnyshoME5RSxn5iQy-TWeFbOA?usp=sharing) in case anyone wants to play. 

- [x] Replace the image links

### CODE REVIEW

File: docs/source/semantic_segmentation.mdx
Comment (nateraw) [inline docs/source/semantic_segmentation.mdx:4]:
```suggestion
a wide variety of applications enabled by these datasets such as background removal from images, stylizing
```

File: docs/source/semantic_segmentation.mdx
Comment (nateraw) [inline docs/source/semantic_segmentation.mdx:5]:
```suggestion
images, or scene understanding for autonomous driving. This guide will show you how to apply transformations
```

File: docs/source/semantic_segmentation.mdx
Comment (nateraw) [inline docs/source/semantic_segmentation.mdx:6]:
```suggestion
to an image segmentation dataset.
```

File: docs/source/semantic_segmentation.mdx
Comment (nateraw) [inline docs/source/semantic_segmentation.mdx:59]: Maybe actually define this function instead of linking to it above and never defining it?

File: docs/source/semantic_segmentation.mdx
Comment (nateraw) [inline docs/source/semantic_segmentation.mdx:161]:
```suggestion
to learn how to train a semantic segmentation model using the techniques shown in this guide. 
```

**Review by nateraw — COMMENTED** [review]
Some quick notes

File: docs/source/semantic_segmentation.mdx
Comment (nateraw) [reply docs/source/semantic_segmentation.mdx:59]: idk fine either way I suppose. It's quite long

Comment (sayakpaul) [reply docs/source/semantic_segmentation.mdx:59]: Yeah, because of the length, I decided not to include it.

File: docs/source/semantic_segmentation.mdx
Comment (osanseviero) [inline docs/source/semantic_segmentation.mdx:50]:
```suggestion
a [color palette](https://github.com/tensorflow/models/blob/master/research/deeplab/utils/get_dataset_colormap.py):
```

File: docs/source/semantic_segmentation.mdx
Comment (osanseviero) [inline docs/source/semantic_segmentation.mdx:83]: Some people might not know what `albumentations` is, so maybe add a few words about what it is

**Review by osanseviero — APPROVED** [review]
Very cool!

File: docs/source/semantic_segmentation.mdx
Comment (nateraw) [reply docs/source/semantic_segmentation.mdx:50]: is it not better to have permalink here in case codebase changes?

Comment (osanseviero) [reply docs/source/semantic_segmentation.mdx:50]: Good point!

File: docs/source/semantic_segmentation.mdx
Comment (stevhliu) [inline docs/source/semantic_segmentation.mdx:36]: I think it'd be nice to put this code in a code block so users can easily copy it (same with the code below)

File: docs/source/semantic_segmentation.mdx
Comment (stevhliu) [inline docs/source/semantic_segmentation.mdx:1]:
```suggestion
# Semantic segmentation
```

File: docs/source/semantic_segmentation.mdx
Comment (stevhliu) [inline docs/source/semantic_segmentation.mdx:50]:
```suggestion
We can also add a [color palette](https://github.com/tensorflow/models/blob/3f1ca33afe3c1631b733ea7e40c294273b9e406d/research/deeplab/utils/get_dataset_colormap.py#L51) on the segmentation mask and overlay it on top of the original image to visualize the dataset:
```

I think its clearer/stronger if we lead with exactly what we're going to do. Then we can remove the sentence right after it :)

File: docs/source/semantic_segmentation.mdx
Comment (stevhliu) [inline docs/source/semantic_segmentation.mdx:119]:
```suggestion
You can verify the transformation worked by indexing into the `pixel_values` and `label` of an example:
```

File: docs/source/semantic_segmentation.mdx
Comment (stevhliu) [inline docs/source/semantic_segmentation.mdx:132]:
```suggestion
In this guide, you have used `albumentations` for augmenting the dataset. It's also possible to use `torchvision` to apply some similar transforms.
```

File: docs/source/semantic_segmentation.mdx
Comment (stevhliu) [inline docs/source/semantic_segmentation.mdx:160]: I think the transition here can be a little smoother. Maybe something like:

Now that you know how to process a dataset for semantic segmentation, learn [how to train a semantic segmentation model](link to doc) and use it for inference.

**Review by stevhliu — APPROVED** [review]
Thanks for adding this, super nice!

File: docs/source/semantic_segmentation.mdx
Comment (sayakpaul) [reply docs/source/semantic_segmentation.mdx:160]: I agree it's a better transition. Maybe we could follow something similar for the rest of the related guides in `datasets`?

**Review by osanseviero — APPROVED** [review]
Very nice, thanks a lot! :fire: 

In the future feel free to mark comments as resolved if you have fixed those already :) Makes it easier to see what has been fixed/updated for the reviewer

**Comment by sayakpaul** [comment]
Thanks @osanseviero. Am I good to merge?

**Comment by osanseviero** [comment]
I would wait for a second approval just in case :)

**Comment by sayakpaul** [comment]
Sure :)

File: docs/source/semantic_segmentation.mdx
Comment (sayakpaul) [reply docs/source/semantic_segmentation.mdx:26]: Sure, but note that the dataset only has `train` split. But I guess it's fine load that split to make the code bit less verbose.

File: docs/source/semantic_segmentation.mdx
Comment (stevhliu) [reply docs/source/semantic_segmentation.mdx]: For sure, that sounds good to me :)

File: docs/source/semantic_segmentation.mdx
Comment (sayakpaul) [reply docs/source/semantic_segmentation.mdx:140]: Nevermind, I figured it out. 

Here's the PR: https://huggingface.co/datasets/huggingface/documentation-images/discussions/8. 

Once that is merged, I will update the links here and merge this PR. Sounds good? @lhoestq @osanseviero?

Comment (lhoestq) [reply docs/source/semantic_segmentation.mdx:140]: Nice ! thanks :)

**Comment by sayakpaul** [comment]
Merging since the images have been pushed as LFS files ([PR](https://huggingface.co/datasets/huggingface/documentation-images/discussions/8)).

**Pull request merged**
